core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponent.scala

```diff
@@ -26,10 +26,29 @@ trait AttributeRecord[OWNER_ID] {
   val valueEntityRef: Option[Long]
   val listIndex: Option[Int]
   val listLength: Option[Int]
+
+  def toComparable: ComparableAttributeRecord[OWNER_ID]
 }
 
 trait AttributeScratchRecord[OWNER_ID] extends AttributeRecord[OWNER_ID] {
   val transactionId: String
+  override def toComparable = throw new RawlsException("There is no equivalent comparable type for AttributeScratchRecords")
```

Comment:
Alternative: a trait which extends from only AttributeRecords that we will want to compare, so we don't have to do this